lib/convection/model/exceptions.rb

```diff
@@ -0,0 +1,17 @@
+ExcessiveResourcesError = Class.new( StandardError)
+ExcessiveResourceNameError = Class.new( StandardError)
+ExcessiveMappingsError = Class.new( StandardError)
+ExcessiveMappingAttributesError = Class.new( StandardError)
+ExcessiveMappingNameError = Class.new( StandardError)
+ExcessiveMappingAttributeNameError = Class.new( StandardError)
+ExcessiveParametersError = Class.new( StandardError)
+ExcessiveParameterNameError = Class.new( StandardError)
+ExcessiveParameterBytesizeError = Class.new( StandardError)
+ExcessiveOutputsError = Class.new( StandardError)
+ExcessiveOutputNameError = Class.new( StandardError)
+ExcessiveDescriptionError = Class.new( StandardError)
+ExcessiveTemplateSizeError = Class.new( StandardError)
+
+def LimitExceededError(value, limit, errorClass)
+  raise errorClass, "Value #{value} exceeds Limit #{limit}"
+end
```

Comment:
It also might make sense to subclass all of the specific limit exceptions with a more general limit exception so that we can rescue on general limit errors.